macros/field_definitions/get_conversion_columns.sql

```diff
@@ -17,135 +17,203 @@ You may obtain a copy of the Snowplow Personal and Academic License Version 1.0
-    ,SUM(CASE WHEN {{ conv_object['condition'] }} THEN coalesce({{ conv_object['value'] }}, {{ conv_object.get('default_value', 0) }}) ELSE 0 END) AS cv_{{ conv_object['name'] }}_total
-    {%- endif %}
-    ,MIN(CASE WHEN {{ conv_object['condition'] }} THEN derived_tstamp ELSE null END) AS cv_{{ conv_object['name'] }}_first_conversion
-    ,CAST(MAX(CASE WHEN {{ conv_object['condition'] }} THEN 1 ELSE 0 END) AS {{ type_boolean() }}) AS cv_{{ conv_object['name'] }}_converted
+    {% if var('snowplow__enable_conversions', false) %}
+      ,COUNT(CASE WHEN cv_type = '{{ conv_object['name'] }}' THEN 1 ELSE null END) AS cv_{{ conv_object['name'] }}_volume
+      {%- if conv_object.get('list_events', false) %}
+      ,ARRAYAGG(CASE WHEN cv_type = '{{ conv_object['name'] }}'  THEN event_id ELSE null END) WITHIN GROUP (ORDER BY cv_tstamp, dvce_created_tstamp, event_id) AS cv_{{ conv_object['name'] }}_events
+      {%- endif -%}
+      {%- if conv_object.get('value', none) %}
+      ,ARRAYAGG(CASE WHEN cv_type = '{{ conv_object['name'] }}'  THEN coalesce(cv_value,{{ conv_object.get('default_value', 0) }})  ELSE null END) WITHIN GROUP (ORDER BY cv_tstamp, dvce_created_tstamp, event_id) AS cv_{{ conv_object['name'] }}_values
+      ,SUM(CASE WHEN cv_type = '{{ conv_object['name'] }}'  THEN coalesce(cv_value, {{ conv_object.get('default_value', 0) }}) ELSE 0 END) AS cv_{{ conv_object['name'] }}_total
+      {%- endif %}
+      ,MIN(CASE WHEN cv_type = '{{ conv_object['name'] }}'  THEN cv_tstamp ELSE null END) AS cv_{{ conv_object['name'] }}_first_conversion
+      ,CAST(MAX(CASE WHEN cv_type = '{{ conv_object['name'] }}'  THEN 1 ELSE 0 END) AS {{ type_boolean() }}) AS cv_{{ conv_object['name'] }}_converted
+    {% else %}
+      ,COUNT(CASE WHEN {{ conv_object['condition'] }} THEN 1 ELSE null END) AS cv_{{ conv_object['name'] }}_volume
+      {%- if conv_object.get('list_events', false) %}
+      ,ARRAYAGG(CASE WHEN {{ conv_object['condition'] }} THEN event_id ELSE null END) WITHIN GROUP (ORDER BY derived_tstamp, dvce_created_tstamp, event_id) AS cv_{{ conv_object['name'] }}_events
+      {%- endif -%}
+      {%- if conv_object.get('value', none) %}
+      ,ARRAYAGG(CASE WHEN {{ conv_object['condition'] }} THEN coalesce({{ conv_object['value'] }},{{ conv_object.get('default_value', 0) }})  ELSE null END) WITHIN GROUP (ORDER BY derived_tstamp, dvce_created_tstamp, event_id) AS cv_{{ conv_object['name'] }}_values
+      ,SUM(CASE WHEN {{ conv_object['condition'] }} THEN coalesce({{ conv_object['value'] }}, {{ conv_object.get('default_value', 0) }}) ELSE 0 END) AS cv_{{ conv_object['name'] }}_total
+      {%- endif %}
+      ,MIN(CASE WHEN {{ conv_object['condition'] }} THEN derived_tstamp ELSE null END) AS cv_{{ conv_object['name'] }}_first_conversion
+      ,CAST(MAX(CASE WHEN {{ conv_object['condition'] }} THEN 1 ELSE 0 END) AS {{ type_boolean() }}) AS cv_{{ conv_object['name'] }}_converted
+    {% endif %}
   {%- else -%}
```

Comment:
nit: I feel like there's a way to simplify this by setting some variable upfront and then using those rather than basically duplicating all the lines

Reply:
{% set when_condition = "cv_type = '{{ conv_object['name'] }}'"  I don't see how we can work around this due to the = sign in the condition. I don't think if I remove the curlies it will ever get compiled due to the initial " which will make everything a text